### PR TITLE
[Audit][Implementation] - Reduce Redundant Operations

### PR DESCRIPTION
**Pull request by hurali97**
### Details

Part of [Callstack Performance Audit](https://github.com/Expensify/App/issues/33070).

This PR removes the redundant operations like static translation that happens on each loop iteration and this PR also checks beforehand whether the number to parse is a valid phone number and if it is, only then the number is parsed by `parsePhoneNumber`. The reason is that call to `parsePhoneNumber` is costly and should only be executed for the numbers that are valid.

### Fixed Issues

$ https://github.com/Expensify/App/issues/37275
PROPOSAL: https://github.com/Expensify/App/issues/35234#issuecomment-1926911643

### Tests

- [X] Verify that no errors appear in the JS console

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/47336142/f984b6a7-f37c-45ed-8453-665a3c831600

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/47336142/3bf70fe1-7c3d-479b-bea1-7be9f2fea83b

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/47336142/758dbcf2-0260-4281-8e22-f56cbdefeabc

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/47336142/4c82bb26-b9ba-477b-98d4-ec5011dc6ab9

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/47336142/a39eaed6-1428-4795-8f31-daa33fedff45

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/47336142/9f6a4787-1c39-44dd-86ea-633a434631d9

</details>

### CODE REVIEW

File: src/libs/PersonalDetailsUtils.ts
Comment (adhorodyski) [inline src/libs/PersonalDetailsUtils.ts:32]: if I understand we want everything past `MERGED_`, doesn't this return slightly incorrectly without the 1st character right after the `_`?

File: src/libs/PersonalDetailsUtils.ts
Comment (adhorodyski) [inline src/libs/PersonalDetailsUtils.ts:34]: this function is getting called a lot of times, maybe since we default to the `displayName` being returned (I assume that's the hot path) we can check whether to return it first and only then declare the additional `fallbackValue` constant we fallback to? Might save us some memory being allocated.

File: src/libs/PersonalDetailsUtils.ts
Comment (hurali97) [reply src/libs/PersonalDetailsUtils.ts:32]: Correct. I missed that some how. Fixed now 👍

File: src/libs/PersonalDetailsUtils.ts
Comment (hurali97) [reply src/libs/PersonalDetailsUtils.ts:42]: I agree and nice spotting 👍

File: src/libs/PersonalDetailsUtils.ts
Comment (adhorodyski) [reply src/libs/PersonalDetailsUtils.ts:32]: uff good we spotted that hahah

Comment (adhorodyski) [reply src/libs/PersonalDetailsUtils.ts:32]: Might be worth documenting the `substringStartIndex` constant with this little explanation! It'd be valuable to have this information in a comment - what it helps to strip and why.

File: src/libs/PersonalDetailsUtils.ts
Comment (adhorodyski) [inline src/libs/PersonalDetailsUtils.ts:28]: looks like we're missing a new line in between just for the nice formatting

File: src/libs/LocalePhoneNumber.ts
Comment (roryabraham) [inline src/libs/LocalePhoneNumber.ts:23]: I wonder if we could improve this at compile time by creating separate types for `EmailLogin` and `SMSLogin`, and making this function accept `SMSLogin` only. Then remove the check in this low-level function? thoughts?

File: src/libs/PersonalDetailsUtils.ts
Comment (roryabraham) [inline src/libs/PersonalDetailsUtils.ts:28]: can you please reserve `/** */` JSDoc-style comments for:

- function signatures
- propType declarations (needed here for use in storybook)

Otherwise, stick to `//` for one-or-two line comments like this one, or `/* */` for larger block comments that aren't JSDocs?

File: src/libs/PersonalDetailsUtils.ts
Comment (roryabraham) [inline src/libs/PersonalDetailsUtils.ts:38]: Is this really much faster than using `passedPersonalDetails.displayName.replace(CONST.REGEX.MERGED_ACCOUNT_PREFIX, '')`?

I heard recently from @tomekzaw that regexes are known to be much slower in Hermes than in other engines. maybe we can try to address that at the source? I'd love to not have to tip-toe around using super basic regexes in our JS code...

File: src/libs/ReportUtils.ts
Comment (roryabraham) [inline src/libs/ReportUtils.ts:777]: move this to CONST as `DEFAULT_POLICY_ROOM_CHAT_TYPES`

**Review by roryabraham — CHANGES_REQUESTED** [review]
Since this is a performance-related PR, can you please update the PR description to include benchmarks?

File: src/libs/ReportUtils.ts
Comment (hurali97) [reply src/libs/ReportUtils.ts:777]: Only reason to not move this to `CONST` was that we will now have same string in two places and if we update it in one place, we will have to do it in other place as well.

For eg, below is what the `CONST` will look like if I add that:

```js
const CONST = {
   // This doesn't work !!
    DEFAULT_POLICY_ROOM_CHAT_TYPES: [CONST.REPORT.CHAT_TYPE.POLICY_ADMINS, CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE, CONST.REPORT.CHAT_TYPE.DOMAIN_ALL],
   ...otherProperties,
   REPORT: {
        CHAT_TYPE: {
            POLICY_ANNOUNCE: 'policyAnnounce',
            POLICY_ADMINS: 'policyAdmins',
            DOMAIN_ALL: 'domainAll',
            POLICY_ROOM: 'policyRoom',
            POLICY_EXPENSE_CHAT: 'policyExpenseChat',
        },
   }
```

Since the above doesn't work, we will have to add magic strings in `DEFAULT_POLICY_ROOM_CHAT_TYPES`:

```js
const CONST = {
    DEFAULT_POLICY_ROOM_CHAT_TYPES: ['policyAdmins', 'policyAnnounce', 'domainAll'],
   ...otherProperties,
   REPORT: {
        CHAT_TYPE: {
            POLICY_ANNOUNCE: 'policyAnnounce',
            POLICY_ADMINS: 'policyAdmins',
            DOMAIN_ALL: 'domainAll',
            POLICY_ROOM: 'policyRoom',
            POLICY_EXPENSE_CHAT: 'policyExpenseChat',
        },
   }
```

Now, we have same string in two places and each time if we update any, we'll have to update other one as well. Thoughts?

File: src/libs/LocalePhoneNumber.ts
Comment (hurali97) [reply src/libs/LocalePhoneNumber.ts:23]: I am not that much familiar with business logic, so I might fancy some insights. From what I understand, we only have one property that holds either Email or Sms/number. How can we differentiate between them? 

I don't know whether we get this property from backend Or we compile the data on front end. In case of latter, we can check whether the property is an email or sms using the logic we have in this PR and add a new property like `isEmail`. Then in `formatPhoneNumber` we can just check if it's email using the new property, we will early return.

But the issue is we will need to pass the `isEmail` property to the function wherever it's called. And just looking in VScode search, it's ~40 places so a lot of refactor. I guess we might be safe with what we have in this PR because it doesn't introduce such huge refactor.

Did I understand correctly?

File: src/libs/CommonTranslationUtils.ts
Comment (roryabraham) [reply src/libs/CommonTranslationUtils.ts:6]: > The approach that is visible as of now, is to introduce a Map which will hold the translated text for each key and each time we invoke Localize.translateLocal(someKey), we will first check whether it exists in the Map and if it doesn't we translate it and store it

Yes, thanks this sounds like a better approach than essentially doing that same thing in a different lib. There are a number of terms for this approach that I think could be used interchangeably in this context:

- caching
- memoization
- dynamic programming
- probably 42 other terms

but it sounds like we're on the same page 👍🏼

File: src/libs/LocalePhoneNumber.ts
Comment (roryabraham) [reply src/libs/LocalePhoneNumber.ts:23]: > From what I understand, we only have one property that holds either Email or Sms/number. How can we differentiate between them?

We could check _once_ at a higher level if a login is an SMS login and narrow the type accordingly.

Less passionate about changing anything here - I think the most important change I'd like to see done in this PR is to move the memoization into the existing localization lib rather than adding "CommonTranslations"

File: src/libs/PersonalDetailsUtils.ts
Comment (roryabraham) [reply src/libs/PersonalDetailsUtils.ts:35]: > However, using the applied solution startsWith only appears to be slightly faster, so it's a matter of discussion whether we should use it or not, since it's only slightly faster

benchmarks would make it easier to discuss this. How much is "slightly faster"?

> I very much like this approach but it's not straight forward to fix performance issues in Hermes. We can take it as an action item from this PR and then follow up on that and also remove the startsWith from this PR. Wdyt?

Sure. I'm sure it would be difficult to improve regex performance in Hermes, but let's look into it. He might hate me for suggesting this, but it seems like it could be a good task for @fabioh8010, since he's worked on regexes in Hermes in the past.

File: src/libs/ReportUtils.ts
Comment (roryabraham) [reply src/libs/ReportUtils.ts:777]: you could also just add separate const for the duplicated strings above the `const CONST = {...}`, right? Or does that break the const-ness in TS?

**Comment by roryabraham** [comment]
marking this as WIP at least until the caching/memoization/dynamicProgramming has been moved to the existing translation lib

File: src/libs/PersonalDetailsUtils.ts
Comment (fabioh8010) [reply src/libs/PersonalDetailsUtils.ts:34]: Hi @roryabraham @hurali97 It looks like an interesting challenge 😅 And definitely more difficult than the indices flag one.

I would suggest creating a follow-up GH issue so we can take a look at it in the future, when I have more capacity to do so. Also, would be great if we could put all evidence we have that Regexes in Hermes are very slow compared to other engines.

File: src/libs/Localize/index.ts
Comment (roryabraham) [inline src/libs/Localize/index.ts:133]: NAB but I wonder if it would make sense to do the cache set operation async so that it doesn't delay the return of `translatedText`:

```suggestion
        setImmediate(() => translatedValues.set(key, translatedText));
```

File: src/libs/Localize/index.ts
Comment (roryabraham) [inline src/libs/Localize/index.ts:111]: It's probably more efficient to have a separate cache for each locale.

File: src/libs/Localize/index.ts
Comment (hurali97) [reply src/libs/Localize/index.ts]: I think let's not introduce fixes for anything which has not performed worse as of yet. There's a chance that this might introduce any regressions as well 🤔

File: src/libs/PersonalDetailsUtils.ts
Comment (hurali97) [reply src/libs/PersonalDetailsUtils.ts:34]: @fabioh8010 Thank you 👍 

I will create an issue shortly and update it all of the relevant information. Just one thing, I will still keep the following check in the codebase because even if we fix the RegEx performance, we are still trying to replace a substring in a string which doesn't contain it, hence the check for `startsWith` is in place.

```ts
    // If the displayName starts with the merged account prefix, remove it.
    if (displayName.startsWith(CONST.MERGED_ACCOUNT_PREFIX)) {
        displayName = displayName.substring(substringStartIndex);
    }
```

We should only be using replace on a string which contains that substring.

File: src/CONST.ts
Comment (hurali97) [reply src/CONST.ts:53]: We need it as the other one we have includes RegExp and we only need the substring `MERGED_`.

**Comment by hurali97** [comment]
@youssef-lr Can you proceed with the review ? 🙂

File: src/libs/PersonalDetailsUtils.ts
Comment (youssef-lr) [reply src/libs/PersonalDetailsUtils.ts:35]: Not sure why we need the extra variable, why not just use  `displayName = displayName.substring(CONST.MERGED_ACCOUNT_PREFIX.length);`?

Comment (youssef-lr) [reply src/libs/PersonalDetailsUtils.ts:35]: bump on my question above @hurali97 please

**Review by youssef-lr — DISMISSED** [review]
LGTM! Left a minor comment [here](https://github.com/Expensify/App/pull/37200/files#r1516044437). All yours @roryabraham

File: src/libs/Localize/index.ts
Comment (roryabraham) [inline src/libs/Localize/index.ts:138]: Just seeing this line, I think it's weird how we're casting a more specific `TranslationPaths` variable to a plain string. I recommend refactoring this like so:

```diff
diff --git a/src/libs/Localize/index.ts b/src/libs/Localize/index.ts
index 72bb4dd89b..9d31bf27bc 100644
--- a/src/libs/Localize/index.ts
+++ b/src/libs/Localize/index.ts
@@ -1,6 +1,7 @@
 import PropTypes from 'prop-types';
 import * as RNLocalize from 'react-native-localize';
 import Onyx from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
 import Log from '@libs/Log';
 import type {MessageElementBase, MessageTextElement} from '@libs/MessageElement';
 import Config from '@src/CONFIG';
@@ -101,49 +102,35 @@ function translate<TKey extends TranslationPaths>(desiredLanguage: 'en' | 'es' |
  * The data is stored in the following format:
  *
  * {
- *  "name": "Name",
+ *   "en": {
+ *     "name": "Name",
+ *   }
  * }
  *
  * Note: We are not storing any translated values for phrases with variables,
  * as they have higher chance of being unique, so we'll end up wasting space
  * in our cache.
  */
-const TRANSLATED_VALUES_EN = new Map<string, string>();
-const TRANSLATED_VALUES_ES = new Map<string, string>();
-const TRANSLATED_VALUES_ES_ES = new Map<string, string>();
-const TRANSLATED_VALUES_ES_ONFIDO = new Map<string, string>();
-
-/**
- * Returns the map for the given locale.
- */
-function getTranslatedValuesMap(locale: BaseLocale) {
-    switch (locale) {
-        case CONST.LOCALES.ES_ES:
-            return TRANSLATED_VALUES_ES_ES;
-        case CONST.LOCALES.ES_ES_ONFIDO:
-            return TRANSLATED_VALUES_ES_ONFIDO;
-        case CONST.LOCALES.ES:
-            return TRANSLATED_VALUES_ES;
-        case CONST.LOCALES.DEFAULT:
-        default:
-            return TRANSLATED_VALUES_EN;
-    }
-}
+const translationCache = new Map<ValueOf<typeof CONST.LOCALES>, Map<TranslationPaths, string>>(
+    Object.values(CONST.LOCALES).reduce((cache, locale) => {
+        cache.push([locale, new Map()]);
+        return cache;
+    }, [] as Array<[ValueOf<typeof CONST.LOCALES>, Map<TranslationPaths, string>]>),
+);
 
 /**
  * Uses the locale in this file updated by the Onyx subscriber.
  */
 function translateLocal<TKey extends TranslationPaths>(phrase: TKey, ...variables: PhraseParameters<Phrase<TKey>>) {
     const preferredLocale = BaseLocaleListener.getPreferredLocale();
-    const key = `${phrase}`;
     const isVariablesEmpty = variables.length === 0;
 
-    // Get the map for the preferred locale
-    const map = getTranslatedValuesMap(preferredLocale);
+    // Get the cache for the preferred locale
+    const cacheForLocale = translationCache.get(preferredLocale);
 
     // Directly access and assign the translated value from the cache, instead of
     // going through map.has() and map.get() to avoid multiple lookups.
-    const valueFromCache = map.get(key);
+    const valueFromCache = cacheForLocale?.get(phrase);
 
     // If the phrase is already translated, return the translated value
     if (valueFromCache) {
@@ -154,7 +141,7 @@ function translateLocal<TKey extends TranslationPaths>(phrase: TKey, ...variable
     // We don't want to store translated values for phrases with variables
     if (isVariablesEmpty) {
         // We set the translated value in the cache
-        map.set(key, translatedText);
+        cacheForLocale?.set(phrase, translatedText);
     }
     return translatedText;
 }

```

File: src/libs/PersonalDetailsUtils.ts
Comment (roryabraham) [inline src/libs/PersonalDetailsUtils.ts:42]: NAB but I don't think this comment makes much sense without the context of the code that was here before, so I think it would be fine to remove it.

**Review by roryabraham — CHANGES_REQUESTED** [review]
Otherwise, LGTM. though I do want to wait for @fedirjh to review, test, and complete the checklist before we merge this

File: src/libs/PersonalDetailsUtils.ts
Comment (hurali97) [reply src/libs/PersonalDetailsUtils.ts:35]: @youssef-lr I added the extra variable just for better readability. I can remove the variable and add the comment instead 👍

File: src/libs/Localize/index.ts
Comment (hurali97) [reply src/libs/Localize/index.ts]: @roryabraham Thank you for simplifying it 🙏

**Comment by roryabraham** [comment]
Cool, now just waiting for @fedirjh to complete the checklist, which [he said he will do soon](https://expensify.slack.com/archives/C05LX9D6E07/p1710171738756239?thread_ts=1710170978.313699&cid=C05LX9D6E07).

File: src/libs/Localize/index.ts
Comment (roryabraham) [reply src/libs/Localize/index.ts:122]: good catch @fedirjh, 100% agree

Comment (hurali97) [reply src/libs/Localize/index.ts:122]: @fedirjh Whoops, nice one 👍 I will look into this today !!

**Comment by hurali97** [comment]
@fedirjh I have addressed your feedback regarding moving cache to `translate` function. Can you re-review the implementation?

Also, below we can see that we save around ~1344 times translating some phrases that's already translated. Again, thanks for highlighting the right place for the cache 💯 


Screenshot 2024-03-12 at 2 27 12 PM

File: src/libs/Localize/index.ts
Comment (roryabraham) [reply src/libs/Localize/index.ts:97]: > If that's the case, why don't we modify the structure of the translations to return a map instead of a plain object in this file?

That makes sense to me. I'm also a bit confused by why `translate` is slow, since it should just be a direct lookup in an object?

Comment (hurali97) [reply src/libs/Localize/index.ts:97]: @fedirjh I will look into this today and see whether changing translation structure to Map performs better than what we have now 👍

Comment (hurali97) [reply src/libs/Localize/index.ts:97]: @fedirjh @roryabraham  Here is an update on how things looks like with:

- translations with object structure ( what we have in main )
- translations with Map structure ( the proposed change by @fedirjh  )
- translations with object structure with cache applied ( what we have right now in this PR )

Below is a video attached which shows the Hermes traces on a heavy account ( Release + Android real device ). For the same section of the trace, we observe that:

- translations with object structure ( before translations Tab in video ) has 11 calls to translate method
- translations with map structure ( after translations with Map Tab in video ) has 5 calls to translate method
- translations with object structure with cache applied ( after translations with cache Tab in video ) has 3 calls to translate method

https://github.com/Expensify/App/assets/47336142/9cfa68cb-7d61-4ce4-be02-f61883331f09


One thing to note here is that "I only created a Map of `en` translations with only 300 entries, whereas we have around 2000 entries in object structure." to test the map structure but since it had more calls than cache implemented, it's safe to say that it's not worth to change object to map.

Below is another video, where we can see the cache implemented and see that how continuously we are accessing the huge translations object to get the desired value for the key. I just want to emphasise that since the cache Map is marginally small as compared to the translations object and that's why we have so less recordings of the `translate` function in Hermes trace, when cache is enabled. It's to note that the `translate` function will still be called but since we are returning from cache and it's quick enough to not get caught in the Hermes trace and the times it get caught was 3.

https://github.com/Expensify/App/assets/47336142/29ea54b5-66ed-4ed8-87c4-9d7607c99860

File: src/libs/Localize/index.ts
Comment (hurali97) [reply src/libs/Localize/index.ts:120]: @fedirjh Oh sorry, I missed this. I will take a look now 👍

Comment (hurali97) [reply src/libs/Localize/index.ts]: @fedirjh I have incorporated this change and also added the JS docs comments for the new `getTranslatedPhrase` function. Thank you for the feedback, appreciate it.

**Review by fedirjh — APPROVED** [review]
Looks good to me.

cc @roryabraham

**Review by roryabraham — APPROVED** [review]

**Pull request merged**
